frb_example/with_flutter/rust/src/bridge_generated.rs

```diff
@@ -316,188 +223,488 @@ where
+        scale: f64,
+        num_threads: i32,
+    ) {
+        wire_draw_mandelbrot_impl(port_, image_size, zoom_point, scale, num_threads)
+    }
+    #[wasm_bindgen]
+    pub fn wire_passing_complex_structs(port_: MessagePort, root: JsValue) {
+        wire_passing_complex_structs_impl(port_, root)
+    }
+    #[wasm_bindgen]
+    pub fn wire_returning_structs_with_boxed_fields(port_: MessagePort) {
+        wire_returning_structs_with_boxed_fields_impl(port_)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_memory_test_input_array(port_: MessagePort, input: Box<[u8]>) {
+        wire_off_topic_memory_test_input_array_impl(port_, input)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_memory_test_output_zero_copy_buffer(port_: MessagePort, len: i32) {
+        wire_off_topic_memory_test_output_zero_copy_buffer_impl(port_, len)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_memory_test_output_vec_u8(port_: MessagePort, len: i32) {
+        wire_off_topic_memory_test_output_vec_u8_impl(port_, len)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_memory_test_input_vec_of_object(port_: MessagePort, input: JsValue) {
+        wire_off_topic_memory_test_input_vec_of_object_impl(port_, input)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_memory_test_output_vec_of_object(port_: MessagePort, len: i32) {
+        wire_off_topic_memory_test_output_vec_of_object_impl(port_, len)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_memory_test_input_complex_struct(port_: MessagePort, input: JsValue) {
+        wire_off_topic_memory_test_input_complex_struct_impl(port_, input)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_memory_test_output_complex_struct(port_: MessagePort, len: i32) {
+        wire_off_topic_memory_test_output_complex_struct_impl(port_, len)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_deliberately_return_error(port_: MessagePort) {
+        wire_off_topic_deliberately_return_error_impl(port_)
+    }
+    #[wasm_bindgen]
+    pub fn wire_off_topic_deliberately_panic(port_: MessagePort) {
+        wire_off_topic_deliberately_panic_impl(port_)
+    }
+    impl Wire2Api<String> for String {
+        fn wire2api(self) -> String {
+            self
         }
     }
-}
 
-impl Wire2Api<Size> for wire_Size {
-    fn wire2api(self) -> Size {
-        Size {
-            width: self.width.wire2api(),
-            height: self.height.wire2api(),
+    impl Wire2Api<Vec<Size>> for JsValue {
+        fn wire2api(self) -> Vec<Size> {
+            self.unchecked_into::<JsArray>()
+                .iter()
+                .map(Wire2Api::wire2api)
+                .collect()
         }
     }
-}
-
-impl Wire2Api<TreeNode> for wire_TreeNode {
-    fn wire2api(self) -> TreeNode {
-        TreeNode {
-            name: self.name.wire2api(),
-            children: self.children.wire2api(),
+    impl Wire2Api<Vec<TreeNode>> for JsValue {
+        fn wire2api(self) -> Vec<TreeNode> {
```

Comment:
A high-level quick question to check whether I understand the code correctly:

So, this implementation uses `JsValue` and js arrays as wire types. This is contrary to the "io" implementation, where we use the raw C structs, pointers, etc, as wire types.

That makes the code quite safer I guess :) Because almost no `unsafe` blocks here (JsValue itself seems to be safe)!

Only worry: Is the performance good enough?

Reply:
Performance would be very interesting to analyze. I know that `wasm_bindgen_serde` was created to solve the performance problems of `serde_json` using methods not too different from ours, but it turned out to vary greatly between workloads. I assume that for larger objects, string deserialization will become cheaper.

Reply:
Anyway, since if a programmer needs to communicate between flutter web and rust, he will probably use this approach, then we are ok - having the same approach and same performance :)